Fix decorator return types

Kostia-K opened the pull request:
Some decorators do not preserve the type of the decorated function. Code sample in the related issue.

<!--
Hi, thanks for submitting a Pull Request. We appreciate it.

Please, fill in all the required information
to make our review and merging processes easier.

Cheers!
-->

## Related issues

- Fixes https://github.com/typeddjango/django-stubs/issues/1464

<!--
Mark what issues this Pull Request closes or references.

Format is:
- Closes #issue-number
- Refs #issue-number

Example. Refs #0
Documentation: https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/
-->

<!--
If you have any feedback, just write it here.

It can be whatever you want!
-->

Kostia-K committed:
Add decorator return types

sobolevn commented on django-stubs/views/decorators/http.pyi:
```diff
@@ -11,6 +11,6 @@ require_GET: Callable[[_F], _F]
 require_safe: Callable[[_F], _F]
 
-def condition(etag_func: Callable | None = ..., last_modified_func: Callable | None = ...) -> Callable: ...
+def condition(etag_func: Callable | None = ..., last_modified_func: Callable | None = ...) -> Callable[[_F], _F]: ...
```
Can you please also type `etag_func` and `last_modified_func` while we are at it? https://docs.djangoproject.com/en/4.2/topics/conditional-view-processing/#the-condition-decorator

Kostia-K replied:
Yes, good idea. I've updated both according to the docs.
Also, in the same commit, changed sensitive variables and post parameters from `Any` to `str` because that's what Django expects.

sobolevn reviewed (commented):
Thank you! Good catch

Kostia-K committed:
Update decorator parameter types

- Sensitive variables and post parameters are always strings
- Update `etag_func` and `last_modified_func` return types based on
  Django docs https://docs.djangoproject.com/en/4.2/topics/conditional-view-processing/#the-condition-decorator

"The function passed last_modified_func should return a standard
datetime value specifying the last time the resource was modified, or
None if the resource doesn’t exist. The function passed to the etag
decorator should return a string representing the ETag for the resource,
or None if it doesn’t exist."

sobolevn reviewed (approved):
Thank you! 👍
